### ISSUE TITLE
hidpi support

### ISSUE THREAD

**Issue by jkall** [jkall/qgis-midvatten-plugin#248]
Some parts of gui are nor suitable for hidpi displays. See screenshot:
![image](https://user-images.githubusercontent.com/6205804/32442841-147d31f2-c2fd-11e7-8cdf-aa36102b103b.png)

Enable better support by:
- avoid hardcoded pixel values
- force resize of dialogs
- only svg icons

further reading:
https://www.itopen.it/qgis-and-qt-getting-ready-for-hidpi-screens/

**Comment by jkall** [comment]
I can not reproduce this issue when using the qgis3 branch on virtual win10 and ubuntu 18.04. 
Hopefully most of the hidpi problems are fixed by using qt5 and matplotlib >=2?

**Comment by jkall** [comment]
No, unfortunately. This is a major issue. Several parts of ui are completely unusable.
![image](https://user-images.githubusercontent.com/6205804/53597068-847bf380-3ba1-11e9-8138-95f11f0ac743.png)

**Comment by HenrikSpa** [comment]
The section plot ui is redone (waiting eagerly for the verdict) in branch simon-merge. This issue should be fixed for section plot.

**Comment by jkall** [comment]
Super!
 Just tested quickly and most parts of gui looks good. Although the font size within matplotlib plot gets small. I guess font size can be edited in the plot settings string (I have not tested).

![screenshot from 2019-03-07 15-13-04](https://user-images.githubusercontent.com/6205804/53962856-5ab85480-40ec-11e9-8227-72e5fe40a054.png)
![screenshot from 2019-03-07 15-13-23](https://user-images.githubusercontent.com/6205804/53962860-5e4bdb80-40ec-11e9-9dd9-fac711d9b771.png)

**Comment by HenrikSpa** [comment]
Yes the fontsize is set in the templates.
I've worked from home and adjusted the sizes to work with a very small screen and low resolution. Maybe it could be solved by having two "default" templates with different settings so the user can choose the best for the moment.

**Comment by jkall** [comment]
Maybe a set of templates where font size, markers (and legend symbology size) are adjusted to be useful on various dpi? Or should it be on various pixel density?
A quick search gave that pixel density may be, at least, ranging from 96 to 384 ppi.  
(https://www.eizo.be/knowledge/monitor-expertise/understanding-pixel-density/)  

Suggeted templates:
(96 ppi)
144 ppi
192 ppi
288 ppi
(384 ppi)

**Comment by jkall** [comment]
Seems to be less of a problem nowadays. Probably thanks to general hidpi enhancements in Qt and desktops in general.
